.gitignore

```diff
@@ -2,3 +2,4 @@
 **/*.jar
 **/*.zip
 **/*.tar.gz
+.idea/
```

Comment:
It's the IntelliJ IDEA project configuration, something like `.settings` for a Eclipse project. I think it's common to clone a GitHub repo into a new IDEA project, and we definitely don't want to commit the files under `.idea/`.